### PR TITLE
Fix bugs introduced with memcpy

### PR DESCRIPTION
**Pull request by kainino0x**
Fixes a bunch of tests broken in #616.

- now correctly copy 0 bytes instead of attempting to copy all remaining elements in the source array.
- fix incorrect assertion
- typo

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

**Review by austinEng — APPROVED** [review]
LGTM!

File: src/webgpu/api/operation/vertex_state/correctness.spec.ts [hunk 1]
```diff
@@ -465,7 +465,7 @@ struct VSOutputs {
           ((index * componentCount + component) * expectedComponentSize) %
           data.expectedData.byteLength;
         memcpy(
-          { src: data.expectedData, start: sourceExpectedOffset, length: vertexComponentSize },
+          { src: data.expectedData, start: sourceExpectedOffset, length: expectedComponentSize },
```
Comment (kainino0x) [inline src/webgpu/api/operation/vertex_state/correctness.spec.ts:468]: Typo introduced here https://github.com/gpuweb/cts/pull/616/files#diff-3a70b4c82a015fca142799eb03a501201a21d682efaafea74040d095873bea04R468

**Review by austinEng — APPROVED** [review]
still LGTM

**Review by toji — APPROVED** [review]
LGTM! `!length` getting caught on 0 has bitten me before too. :P

**Comment by kainino0x** [comment]
whoops, it failed to automerge because of a conflict.

**Pull request merged**